TPP.Core/Commands/Definitions/OperatorCommands.cs

```diff
@@ -18,62 +19,88 @@ public class StopToken
             {
                 Aliases = new[] { "adjustpokeyen" },
                 Description = "Operators only: Add or remove pokeyen from an user. " +
                               "Arguments: p<amount>(can be negative) <user> <reason>"
             },
-            new Command("tokensadjust", AdjustTokens)
+            new Command("tokensadjust", AdjustTokens, UserGroup.Operator)
             {
                 Aliases = new[] { "adjusttokens" },
                 Description = "Operators only: Add or remove tokens from an user. " +
                               "Arguments: t<amount>(can be negative) <user> <reason>"
             },
-            new Command("transferbadge", TransferBadge)
+            new Command("transferbadge", TransferBadge, UserGroup.Operator)
             {
                 Description = "Operators only: Transfer badges from one user to another user. " +
                               "Arguments: <gifter> <recipient> <pokemon> <number of badges>(Optional) <reason>"
             },
-            new Command("createbadge", CreateBadge)
+            new Command("createbadge", CreateBadge, UserGroup.Operator)
             {
                 Description = "Operators only: Create a badge for a user. " +
                               "Arguments: <recipient> <pokemon> <number of badges>(Optional)"
             },
-        }.Select(cmd => cmd.WithCondition(
-            canExecute: ctx => IsOperator(ctx.Message.User),
-            ersatzResult: new CommandResult { Response = "Only operators can use that command" }));
-
-        private bool IsOperator(User user) => _operatorNamesLower.Contains(user.SimpleName);
+            new Command("setoperator", setOperator, UserGroup.Operator)
+            {
+                Aliases = new [] { "op", "operator"},
+                Description = "Operators only: Give other users operator status. Removes other roles. " +
+                              "Arguments: <user> <user> ..."
+            },
+            new Command("setmoderator", setModerator, UserGroup.Operator)
+            {
+                Aliases = new [] { "mod", "moderator"},
+                Description = "Operators only: Give other users moderator status. Removes other roles. " +
+                              "Arguments: <user> <user> ..."
+            },
+            new Command("settrusted", setTrusted, UserGroup.Operator)
+            {
+                Aliases = new [] { "trust", "trusted" },
+                Description = "Operator only: Give other users trusted status. Removes modteam roles. " +
+                              "Arguments: <user> <user> ..."
+            },
+            new Command("setmusicteam", setMusicTeam, UserGroup.Operator)
+            {
+                Aliases = new [] { "musicteam" },
+                Description = "Operator only: Give other users music team status. Removes modteam roles. " +
+                              "Arguments: <user> <user> ..."
+            },
```

Comment:
I had set it up this way so that assigning one role would automatically remove other roles that weren't needed E.G. setting someone as moderator removes the now redundant trusted, musicteam roles in the case of a promotion. It also removes the operator role in the event of demotion, simplifying the entire interaction to one command call.

Doing the same in an [addrank <group>] style command I feel would require a less than ideal number of if statements and be messy.

I can change the output to show the modified user(s) updated ranks, however it would still require another call of removerank <group> to enforce users only having 1 of a set of hierarchic ranks.

Reply:
I see what you mean. But we already have a metric shit ton of commands, so it feels like having the commands' implementation deal with a bit more stuff themselves in favor of only having to remember one commands feels preferable for me. Maybe other people in #new-core feel different about this